src/main/java/ch/njol/skript/effects/EffPlaySound.java

```diff
@@ -51,182 +50,127 @@
-							"[(and|at|with) pitch %-number%] [(to|for) %players%] [(at|from) %-locations%]");
-		}
+		Skript.registerEffect(EffPlaySound.class,
+				"play sound[s] %strings% [(in|from) %-soundcategory%] " +
+						"[(at|with) volume %-number%] [(and|at|with) pitch %-number%] at %locations% [(to|for) %-players%]",
+				"play sound[s] %strings% [(in|from) %-soundcategory%] " +
+						"[(at|with) volume %-number%] [(and|at|with) pitch %-number%] [(to|for) %players%] [(at|from) %-locations%]");
 	}
 
-	@SuppressWarnings("null")
+	@SuppressWarnings("NotNullFieldNotInitialized")
 	private Expression<String> sounds;
-	@Nullable
-	private Expression<SoundCategory> category;
-	@Nullable
-	private Expression<Number> volume, pitch;
-	@Nullable
-	private Expression<Location> locations;
-	@Nullable
-	private Expression<Player> players;
+	private @Nullable Expression<SoundCategory> category;
+	private @Nullable Expression<Number> volume;
+	private @Nullable Expression<Number> pitch;
+	private @Nullable Expression<Location> locations;
+	private @Nullable Expression<Player> players;
 
 	@Override
-	@SuppressWarnings({"unchecked", "null"})
+	@SuppressWarnings("unchecked")
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		sounds = (Expression<String>) exprs[0];
-		if (SOUND_CATEGORIES_EXIST) {
-			category = (Expression<SoundCategory>) exprs[1];
-			volume = (Expression<Number>) exprs[2];
-			pitch = (Expression<Number>) exprs[3];
-			if (matchedPattern == 0) {
-				locations = (Expression<Location>) exprs[4];
-				players = (Expression<Player>) exprs[5];
-			} else {
-				players = (Expression<Player>) exprs[4];
-				locations = (Expression<Location>) exprs[5];
-			}
+		category = (Expression<SoundCategory>) exprs[1];
+		volume = (Expression<Number>) exprs[2];
+		pitch = (Expression<Number>) exprs[3];
+		if (matchedPattern == 0) {
+			locations = (Expression<Location>) exprs[4];
+			players = (Expression<Player>) exprs[5];
 		} else {
-			volume = (Expression<Number>) exprs[1];
-			pitch = (Expression<Number>) exprs[2];
-			if (matchedPattern == 0) {
-				locations = (Expression<Location>) exprs[3];
-				players = (Expression<Player>) exprs[4];
-			} else {
-				players = (Expression<Player>) exprs[3];
-				locations = (Expression<Location>) exprs[4];
-			}
+			players = (Expression<Player>) exprs[4];
+			locations = (Expression<Location>) exprs[5];
 		}
+		
 		return true;
 	}
 
 	@Override
-	@SuppressWarnings("null")
-	protected void execute(Event e) {
-		Object category = null;
-		if (SOUND_CATEGORIES_EXIST) {
-			category = SoundCategory.MASTER;
-			if (this.category != null) {
-				category = this.category.getSingle(e);
-				if (category == null)
-					return;
-			}
+	protected void execute(Event event) {
+		SoundCategory category = SoundCategory.MASTER;
+		if (this.category != null) {
+			category = this.category.getSingle(event);
+			if (category == null)
+				return;
 		}
```

Comment:
```suggestion
		SoundCategory category = this.category.getOptionalSingle(event).orElse(SoundCategory.MASTER);
```
Same can go for EffStopSound